packages/@uppy/companion-client/src/RequestClient.js

```diff
@@ -16,27 +18,40 @@ function stripSlash (url) {
   return url.replace(/\/$/, '')
 }
 
+
+const authErrorStatusCode = 401
+
+class HttpError extends Error {
```

Comment:
I think we do need to get a hold of the errors we have. There is `NetworkError`, `HttpError`, and `StreamHttpJsonError`. Should probably be possible to have two? It's okay to make breaking changes in `@uppy/utils` (as long as we change it everywhere). Perhaps a single file `errors.ts`, with all our errors + docs when they are used?